### ISSUE TITLE
sendpayment.py zero value change ouput

### ISSUE THREAD

**Issue by CohibAA** [chris-belcher/joinmarket#196]
Related to #168:

If only one utxo is required to satisfy [total_amount](https://github.com/chris-belcher/joinmarket/blob/master/sendpayment.py#L77), and the [value of the utxo](https://github.com/chris-belcher/joinmarket/blob/master/sendpayment.py#L80) is equal to (or less than dust difference also I believe) the [total_amount](https://github.com/chris-belcher/joinmarket/blob/master/sendpayment.py#L77), there will be no change required (or it will be below the bitcoind dust threshold).  This may happen when not sweeping on (probably rare) occasion, resulting in a change address with an output of 0 being used at this point in the code.

The code works and continues to create the raw transaction and attempt to broadcast it, however the transaction will be rejected by the network due to being dust/non-standard (below 546 satoshi) output for the 0 output change address.

This results in a crash of the taker's sendpayment.py after generating the raw transaction if using bitcoind because of the rejection error from sendrawtransaction.  Makers will see a completed transaction with "earnings" in their logs, but the transaction will never be broadcast.  I don't believe this will affect future maker transactions/utxo but I am not certain on that point (would it affect the makers' wallet gap if it happens too many times?).

Possible solution(s):  Check [before this line](https://github.com/chris-belcher/joinmarket/blob/master/sendpayment.py#L81) to see if a) the values are equal, as described in the first paragraph then start the coinjoin [the way a sweep does without change](https://github.com/chris-belcher/joinmarket/blob/master/sendpayment.py#L62-#L63) and/or b) check that none of the output values are below the bitcoind dust threshold and adjust the coinjoin to include an additional utxo therefore requiring change (would require to 'start over' with the maker(s), I believe), or just exit with a debug message.

**Comment by chris-belcher** [comment]
OK, understood.

Yield generators don't log any earnings until they see the tx actually confirmed.

I think the solution should be all of those. I think makers already try to find other UTXOs to add, don't remember right now.
Nothing should require starting over the coinjoin because you can calculate everything before you contact any makers.

**Comment by CohibAA** [comment]
> Yield generators don't log any earnings until they see the tx actually confirmed.

Right, I was the taker in this scenario so I wasn't sure what the maker saw for certain.  Thank for clarifying.

> I think makers already try to find other UTXOs to add, don't remember right now.

Again, I was using sendpayment.py as a taker, so it would have needed to add another input from me, to create sufficient change.  When it happened to the transaction I was doing the script attempted the broadcast, received the dust error from bitcoind, then crashed and dumped.  I don't remember the err number and I seem to have accidentally deleted the log from this transaction.  It can be reproduced with some good old fashioned math on the utxos from a wallet depth, though.

The difference between the actual utxo being used from my wallet and the dust threshold was above 546 when I started sendpayment.py and was given the confirmation to proceed with the selected makers.  Once the makers' fees were included, it became below the threshold, and the signed coinjoin transaction was created with the bad (change amount) output, which caused the transaction broadcast error and subsequent crash.
